bindings/c/test/apitester/TesterAtomicOpsCorrectnessWorkload.cpp

```diff
@@ -0,0 +1,368 @@
+			    [](uint64_t val1, uint64_t val2) { return val1 | val2; },
+			    cont);
+			break;
+		case OP_ATOMIC_BIT_XOR:
+			testIntAtomicOp(
+			    FDBMutationType::FDB_MUTATION_TYPE_BIT_XOR,
+			    [](uint64_t val1, uint64_t val2) { return val1 ^ val2; },
+			    cont);
+			break;
+		case OP_ATOMIC_APPEND_IF_FITS: {
+			Value val1 = randomValue();
+			Value val2 = randomValue();
+			testAtomicOp(
+			    FDBMutationType::FDB_MUTATION_TYPE_APPEND_IF_FITS,
+			    val1,
+			    val2,
+			    [](ValueRef val1, ValueRef val2) { return Value(val1) + Value(val2); },
+			    cont);
+			break;
+		}
+		case OP_ATOMIC_MAX:
+			testIntAtomicOp(
+			    FDBMutationType::FDB_MUTATION_TYPE_MAX,
+			    [](uint64_t val1, uint64_t val2) { return std::max(val1, val2); },
+			    cont);
+			break;
+		case OP_ATOMIC_MIN:
+			testIntAtomicOp(
+			    FDBMutationType::FDB_MUTATION_TYPE_MIN,
+			    [](uint64_t val1, uint64_t val2) { return std::min(val1, val2); },
+			    cont);
+			break;
+		case OP_ATOMIC_VERSIONSTAMPED_KEY:
+			testAtomicVersionstampedKeyOp(cont);
+			break;
+		case OP_ATOMIC_VERSIONSTAMPED_VALUE:
+			testAtomicVersionstampedValueOp(cont);
+			break;
+		case OP_ATOMIC_BYTE_MIN: {
+			Value val1 = randomValue();
+			Value val2 = randomValue();
+			testAtomicOp(
+			    FDBMutationType::FDB_MUTATION_TYPE_BYTE_MIN,
+			    val1,
+			    val2,
+			    [](ValueRef val1, ValueRef val2) { return Value(std::min(val1, val2)); },
+			    cont);
+			break;
+		}
+		case OP_ATOMIC_BYTE_MAX: {
+			Value val1 = randomValue();
+			Value val2 = randomValue();
+			testAtomicOp(
+			    FDBMutationType::FDB_MUTATION_TYPE_BYTE_MAX,
+			    val1,
+			    val2,
+			    [](ValueRef val1, ValueRef val2) { return Value(std::max(val1, val2)); },
+			    cont);
+			break;
+		}
+		case OP_ATOMIC_COMPARE_AND_CLEAR:
+			testAtomicCompareAndClearOp(cont);
+			break;
+		}
+	}
+
+	Key getKeySuffix() { return Key(fdb::toBytesRef(fmt::format("_{}_{}", clientId, atomicOpId++))); }
+
+	void testIntAtomicOp(FDBMutationType opType, IntAtomicOpFunction opFunc, TTaskFct cont) {
+		uint64_t intValue1 = Random::get().randomInt(0, 10000000);
+		uint64_t intValue2 = Random::get().randomInt(0, 10000000);
+
+		Value val1 = toByteString(intValue1);
+		Value val2 = toByteString(intValue2);
+		testAtomicOp(
+		    opType,
+		    val1,
+		    val2,
+		    [opFunc](ValueRef val1, ValueRef val2) { return toByteString(opFunc(toUint64_t(val1), toUint64_t(val2))); },
+		    cont);
+	}
+
+	void testAtomicOp(FDBMutationType opType, Value val1, Value val2, AtomicOpFunction opFunc, TTaskFct cont) {
+		Key key(randomKeyName() + getKeySuffix());
+		execTransaction(
+		    // 1. Set the key to val1
+		    [key, val1](auto ctx) {
+			    ctx->tx().set(key, val1);
+			    ctx->commit();
+		    },
+		    [this, opType, opFunc, key, val1, val2, cont]() {
+			    execTransaction(
+			        // 2. Perform the given atomic operation to val2, but only if it hasn't been applied yet, otherwise
```

Comment:
For that I would need to modify the framework to avoid automatic retries, and to propagate the error to the client of execTransaction() (although the latter may already by done).

Reply:
Then let's do it in a separate PR